### PR TITLE
Added first Chapter

### PR DESCRIPTION
**Pull request by mohamedfarith**
I have made the first pull request

### CODE REVIEW

File: app/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -16,8 +16,43 @@
                 <category android:name="android.intent.category.LAUNCHER" />
             </intent-filter>
         </activity>
-        <activity android:name=".Main3Activity" />
-        <activity android:name=".Main4Activity" />
+        <activity
+            android:name=".Activity_of_Learning_Activity"
```
Comment (nmrafiq87) [inline app/src/main/AndroidManifest.xml]: Please follow naming convention for classes

File: app/src/main/java/com/example/admin/learningandroid/Activity_of_Layout_Activity.java [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+import android.support.v7.app.AppCompatActivity;
+import android.os.Bundle;
+
+public class Activity_of_Layout_Activity extends AppCompatActivity {
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.layout_of_layout_activity);
```
Comment (nmrafiq87) [inline app/src/main/java/com/example/admin/learningandroid/Activity_of_Layout_Activity.java]: Meaningful name for layout. Like home_layout

File: app/src/main/java/com/example/admin/learningandroid/Activity_of_Learning_Activity.java [hunk 3]
```diff
@@ -0,0 +1,44 @@
+package com.example.admin.learningandroid;
+
+import android.content.Intent;
+import android.support.v7.app.AppCompatActivity;
+import android.os.Bundle;
+import android.view.View;
+import android.widget.Button;
+
+public class Activity_of_Learning_Activity extends AppCompatActivity {
+    Button activity_lifecycle, tasks, passing_object;
```
Comment (nmrafiq87) [inline app/src/main/java/com/example/admin/learningandroid/Activity_of_Learning_Activity.java]: Do not declare variables on the same line. Use line breaks

File: app/src/main/java/com/example/admin/learningandroid/Activity_of_Learning_Activity_Lifecycle_1.java [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+import android.content.Intent;
+import android.support.v7.app.AppCompatActivity;
+import android.os.Bundle;
+import android.util.Log;
+import android.view.View;
+import android.widget.Button;
+
+public class Activity_of_Learning_Activity_Lifecycle_1 extends AppCompatActivity {
+    String Tag = "activity_1";
+    Button next_activity;
```
Comment (nmrafiq87) [inline app/src/main/java/com/example/admin/learningandroid/Activity_of_Learning_Activity_Lifecycle_1.java]: Follow proper naming convention for class and local variables

**Review by nmrafiq87 — CHANGES_REQUESTED** [review]

File: app/src/main/AndroidManifest.xml (hunk 1 above)
Comment (nmrafiq) [reply app/src/main/AndroidManifest.xml]: The Class Name should not have underscore. And the name ActivityOfLearningActivity does not make a lot of sense. Instead you can use LearningActivity.

File: app/src/main/AndroidManifest.xml [hunk 5]
```diff
@@ -16,8 +16,43 @@
...
+            android:label="Activity_of_Learning_Activity" />
+        <activity
+            android:name=".Activity_of_Learning_Activity_Lifecycle_1"
+            android:label="Activity_of_Learning_Activity_Lifecycle_1" />
+        <activity android:name=".Activity_of_Learning_Activity_Lifecycle_2" />
+        <activity
+            android:name=".Activity_of_Learning_Tasks"
```
Comment (nmrafiq) [inline app/src/main/AndroidManifest.xml]: Same as mentioned above. Use meaningful names without underscore. Use sentence case for eg: LearningLifecycle

File: app/src/main/AndroidManifest.xml [hunk 6]
```diff
@@ -16,8 +16,43 @@
...
+            android:label="Activity_of_Learning_Tasks" />
+        <activity
+            android:name=".Activity_of_Standard_Task"
+            android:label="Activity_of_Standard_Task"
```
Comment (nmrafiq) [inline app/src/main/AndroidManifest.xml]: Same as mentioned above

File: app/src/main/AndroidManifest.xml [hunk 7]
```diff
@@ -16,8 +16,43 @@
...
+            android:launchMode="standard" />
+        <activity
+            android:name=".Activity_of_Single_Top"
```
Comment (nmrafiq) [inline app/src/main/AndroidManifest.xml]: Same for all the activities

File: app/src/main/java/com/example/admin/learningandroid/Activity_of_Learning_Activity.java (hunk 3 above)
Comment (nmrafiq) [reply app/src/main/java/com/example/admin/learningandroid/Activity_of_Learning_Activity.java]: For eg, 
Button btnActivityLifecycle;
Button btnTasks;
Button btnPassingObject;

And if you are declaring a UI component prefix btn/button for Button or txt/text

Comment (nmrafiq) [reply app/src/main/java/com/example/admin/learningandroid/Activity_of_Learning_Activity.java]: Read this https://source.android.com/setup/contribute/code-style and do apply while coding.

Follow field naming conventions
Non-public, non-static field names start with m.

Static field names start with s.

Other fields start with a lower case letter.

Public static final fields (constants) are ALL_CAPS_WITH_UNDERSCORES.

For example:
The below is mentioned in the same document

public class MyClass {
    public static final int SOME_CONSTANT = 42;
    public int publicField;
    private static MyClass sSingleton;
    int mPackagePrivate;
    private int mPrivate;
    protected int mProtected;
}

File: app/src/main/java/com/example/admin/learningandroid/Activity_of_Learning_Activity.java [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.layout_of_activity_learning);
+        activity_lifecycle = (Button) findViewById(R.id.activity_lifecycle);
+        tasks = (Button) findViewById(R.id.tasks);
+        passing_object = (Button) findViewById(R.id.passing_object);
+
+
+        activity_lifecycle.setOnClickListener(new View.OnClickListener() {
+            @Override
```
Comment (nmrafiq) [inline app/src/main/java/com/example/admin/learningandroid/Activity_of_Learning_Activity.java]: Instead of creating multiple listener you can use one 
tasks.setOnClickListener(new View.OnClickListener() {
     @Override
            public void onClick(View view) {
                int id = view.getId();
                // And then use switch case 
           }
        });

File: app/src/main/java/com/example/admin/learningandroid/Activity_of_Learning_Tasks.java [hunk 9]
```diff
@@ -0,0 +1,52 @@
+package com.example.admin.learningandroid;
+
+import android.content.Intent;
+import android.support.v7.app.AppCompatActivity;
+import android.os.Bundle;
+import android.view.View;
+import android.widget.Button;
+
+public class Activity_of_Learning_Tasks extends AppCompatActivity {
+    Button standard_task, single_top, single_task, single_instance;
```
Comment (nmrafiq) [inline app/src/main/java/com/example/admin/learningandroid/Activity_of_Learning_Tasks.java]: Declare variables in separate lines

File: app/src/main/java/com/example/admin/learningandroid/Activity_of_Learning_Tasks.java [hunk 10]
```diff
@@ -0,0 +1,52 @@
...
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.layout_of__learning__tasks);
+
+        standard_task = (Button) findViewById(R.id.standard_task);
+        single_top = (Button) findViewById(R.id.single_top);
+        single_task = (Button) findViewById(R.id.single_task);
+        single_instance = (Button) findViewById(R.id.single_instance);
+
+        standard_task.setOnClickListener(new View.OnClickListener() {
```
Comment (nmrafiq) [inline app/src/main/java/com/example/admin/learningandroid/Activity_of_Learning_Tasks.java]: Single listener for all the buttons.

File: app/src/main/java/com/example/admin/learningandroid/Activity_of_Passing_Object.java [hunk 11]
```diff
@@ -0,0 +1,38 @@
...
+import android.support.v7.app.AppCompatActivity;
+import android.os.Bundle;
+import android.view.View;
+import android.widget.Button;
+import android.widget.EditText;
+
+import java.util.jar.Attributes;
+
+public class Activity_of_Passing_Object extends AppCompatActivity {
+    EditText first_name, last_name;
```
Comment (nmrafiq) [inline app/src/main/java/com/example/admin/learningandroid/Activity_of_Passing_Object.java]: EditText etFirstName;
EditText etLastName;

File: app/src/main/java/com/example/admin/learningandroid/Activity_of_Passing_Object.java [hunk 12]
```diff
@@ -0,0 +1,38 @@
...
+    Button display_name;
+    String f_name, l_name;
```
Comment (nmrafiq) [inline app/src/main/java/com/example/admin/learningandroid/Activity_of_Passing_Object.java]: Same you can use String strFirstName; Use camel case for variables

File: app/src/main/java/com/example/admin/learningandroid/Activity_of_Passing_Object.java [hunk 13]
```diff
@@ -0,0 +1,38 @@
...
+        display_name = (Button) findViewById(R.id.display_name);
+
+        display_name.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                f_name = first_name.getText().toString();
+                l_name = last_name.getText().toString();
+                Name_of_Person name = new Name_of_Person(f_name, l_name);
+                Intent intent = new Intent(Activity_of_Passing_Object.this, Activity_of_Receiving_Object.class);
+                intent.putExtra("Name", name);
```
Comment (nmrafiq) [inline app/src/main/java/com/example/admin/learningandroid/Activity_of_Passing_Object.java]: The extra key 'Name' can be a constant. 
public static final int KEY_NAME = "Name";
intent.putExtra(KEY_NAME, name);

Comment (nmrafiq) [reply app/src/main/java/com/example/admin/learningandroid/Activity_of_Passing_Object.java]: The same declaration 
public static final int KEY_NAME = "Name"
can be kept in a separate class called Constants

File: app/src/main/java/com/example/admin/learningandroid/Activity_of_Receiving_Object.java [hunk 14]
```diff
@@ -0,0 +1,25 @@
...
+    protected void onCreate(Bundle savedInstanceState) {
+
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.layout_of_receiving_object);
+        full_name = (TextView) findViewById(R.id.full_name);
+
+
+        Intent intent = getIntent();
+        Bundle b = intent.getExtras();
+        Name_of_Person p = b.getParcelable("Name");
```
Comment (nmrafiq) [inline app/src/main/java/com/example/admin/learningandroid/Activity_of_Receiving_Object.java]: Use the key from Constants

File: app/src/main/java/com/example/admin/learningandroid/Activity_of_Single_Instance.java [hunk 15]
```diff
@@ -0,0 +1,90 @@
...
+
+import android.content.Intent;
+import android.support.v7.app.AppCompatActivity;
+import android.os.Bundle;
+import android.util.Log;
+import android.view.View;
+import android.widget.Button;
+
+public class Activity_of_Single_Instance extends AppCompatActivity {
+    String Tag = "Single_Instance";
```
Comment (nmrafiq) [inline app/src/main/java/com/example/admin/learningandroid/Activity_of_Single_Instance.java]: this can be static final and the declaration can be like     public static final int SOME_CONSTANT = 42;
Please read https://source.android.com/setup/contribute/code-style

File: app/src/main/res/layout/activity_of__learning__activity__lifecycle_1.xml [hunk 16]
```diff
@@ -0,0 +1,20 @@
+<?xml version="1.0" encoding="utf-8"?>
+<LinearLayout xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:tools="http://schemas.android.com/tools"
+    android:layout_width="match_parent"
+    android:layout_height="match_parent"
+    android:orientation="vertical">
+
+    <TextView
+        android:id="@+id/activity_1"
```
Comment (nmrafiq) [inline app/src/main/res/layout/activity_of__learning__activity__lifecycle_1.xml]: txt_activity_one is a better name. You know when you use in code that the UI component is of type TextView

File: app/src/main/res/values/strings.xml [hunk 17]
```diff
@@ -3,4 +3,37 @@
 
     <!-- TODO: Remove or change this placeholder text -->
     <string name="hello_blank_fragment">Hello blank fragment</string>
+    <string name="this_is_actiivty_1_for_understanding_activity_life_cycle">This is Actiivty 1 for understanding activity life cycle</string>
+    <string name="this_is_activity_1_for_understanding_activity_lifecycle">this is activity 1 for understanding activity lifecycle</string>
```
Comment (nmrafiq) [inline app/src/main/res/values/strings.xml]: use short, simple and understandable string names

File: app/src/main/java/com/example/admin/learningandroid/Activity_of_Single_Instance.java [hunk 18]
```diff
@@ -0,0 +1,90 @@
...
+            }
+        });
+        single_instance.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                Intent intent = new Intent(Activity_of_Single_Instance.this, Activity_of_Single_Instance.class);
+                startActivity(intent);
+            }
+        });
+        standard.setOnClickListener(new View.OnClickListener() {
```
Comment (nmrafiq) [inline app/src/main/java/com/example/admin/learningandroid/Activity_of_Single_Instance.java]: Why these buttons are there in every activity ? It should ideally be there in only one activity

**Review by nmrafiq — CHANGES_REQUESTED** [review]
Please go through this https://source.android.com/setup/contribute/code-style

File: app/src/main/AndroidManifest.xml [hunk 19]
```diff
@@ -16,8 +16,46 @@
...
+            android:launchMode="singleInstance"
+            android:taskAffinity="com.example.affinity" />
+        <activity
+            android:name=".PassingObjectActivity"
+            android:label="Activity_of_Passing_Object" />
+        <activity
+            android:name=".ReceivingObjectActivity"
+            android:label="Activity_of_Receiving_Object" />
+        <activity android:name=".LearningLayouts" />
+        <activity android:name=".RecyclerViewActivity" />
```
Comment (nmrafiq87) [inline app/src/main/AndroidManifest.xml]: Layout is a separate chapter. Highly recommended if you can raise separate PRs for each chapters.

File: app/src/main/java/com/example/admin/learningandroid/LearningLayouts.java [hunk 20]
```diff
@@ -0,0 +1,56 @@
...
+        btnRelative.setOnClickListener(this);
+        btnConstraint.setOnClickListener(this);
+        btnRecyclerView.setOnClickListener(this);
+    }
+
+    @Override
+    public void onClick(View view) {
+        switch (view.getId()) {
+            case R.id.btnRecyclerView:
+                Intent intent = new Intent(LearningLayouts.this, RecyclerViewActivity.class);
```
Comment (nmrafiq87) [inline app/src/main/java/com/example/admin/learningandroid/LearningLayouts.java]: meaningful names. Like intentRecycler

File: app/src/main/java/com/example/admin/learningandroid/SingleTaskActivity.java [hunk 21]
```diff
@@ -0,0 +1,94 @@
...
+    Button singleTopButton2;
+    Button singleTaskButton2;
+    Button singleInstanceButton2;
+    Button standardButton2;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.singletask_layout);
+        singleTopButton2 = findViewById(R.id.singleTopButton2);
```
Comment (nmrafiq87) [inline app/src/main/java/com/example/admin/learningandroid/SingleTaskActivity.java]: Why is this required everywhere ?

File: app/src/main/java/com/example/admin/learningandroid/LearningLayouts.java [hunk 22]
```diff
@@ -0,0 +1,56 @@
...
+    Button btnRelative;
+    Button btnConstraint;
+    Button btnFrame;
+    Button btnRecyclerView;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.learning_layouts);
+        btnLinear = findViewById(R.id.btnLinear);
```
Comment (nmrafiq87) [inline app/src/main/java/com/example/admin/learningandroid/LearningLayouts.java]: for layout names btn_linear. These are not variables.

File: app/src/main/java/com/example/admin/learningandroid/LearningLayouts.java (hunk 20 above)
Comment (nmrafiq87) [reply app/src/main/java/com/example/admin/learningandroid/LearningLayouts.java]: RecyclerView was never a part of the chapter

File: app/src/main/java/com/example/admin/learningandroid/layouts/ModelLoginPage.java [hunk 23]
```diff
@@ -0,0 +1,49 @@
...
+        btnSave.setOnClickListener(this);
+    }
+
+    @Override
+    public void onClick(View view) {
+        String mName = etName.getText().toString();
+        String mMarks = etMarks.getText().toString();
+        String mRegistrationNumber = etRegistrationNumber.getText().toString();
+        int mCount = mName.length();
+        if (mCount <= 3) {
```
Comment (nmrafiq87) [inline app/src/main/java/com/example/admin/learningandroid/layouts/ModelLoginPage.java:35]: you can opt for early return.. 

For eg : 
if(a>b){
// do something ;
return;
}
if (c>d){
// do something return 
}

File: app/src/main/java/com/example/admin/learningandroid/layouts/ModelLoginPage.java [hunk 24]
```diff
@@ -0,0 +1,49 @@
...
+import android.support.v7.app.AppCompatActivity;
+import android.os.Bundle;
+import android.view.View;
+import android.widget.Button;
+import android.widget.EditText;
+import android.widget.Toast;
+
+import com.example.admin.learningandroid.R;
+
+public class ModelLoginPage extends AppCompatActivity implements View.OnClickListener {
```
Comment (nmrafiq87) [inline app/src/main/java/com/example/admin/learningandroid/layouts/ModelLoginPage.java:12]: ModalLoginActivity should be the name

**Review by nmrafiq87 — CHANGES_REQUESTED** [review]
